src/main/java/org/openapitools/api/impl/SessionsApiServiceImpl.java

```diff
@@ -27,27 +26,21 @@ public Response sessionsPost(Session session, SecurityContext securityContext) t
             LoggedInSession
                 loggedInSession =
                 identity.getSessionService().newSessionWithPassword(session.getUsername(), session.getPassword());
-            String token = loggedInSession.getCredential().toString();
-
-            ApiResponseMessage
-                successMsg =
-                new ApiResponseMessage(Response.Status.CREATED.getStatusCode(), "The session is created successfully");
-            return Response.status(Response.Status.CREATED).entity(successMsg).header("Set-Cookie", token).build();
-
-        } catch (NotAuthorizedException e) {
-            ApiResponseMessage
-                errorMsg =
-                new ApiResponseMessage(Response.Status.UNAUTHORIZED.getStatusCode(),
-                                       "Invalid request: " + e.getMessage());
-            return Response.status(Response.Status.UNAUTHORIZED).entity(errorMsg).build();
-        } catch (Exception e) {
-            ApiResponseMessage
-                errorMsg =
-                new ApiResponseMessage(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
-                                       "Unknown error occurs: " + e.getMessage());
-            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(errorMsg).build();
+            String cookie = loggedInSession.getCredential().toString();
+
+            return Response.status(Response.Status.CREATED).entity("The session is created successfully")
+                .header("Set-Cookie", "V=" + cookie)
+                .build();
+
+        } catch (IdentityException e) {
+            switch (e.getError()) {
+                case ACCOUNT_NOT_FOUND:
+                    return Response.status(Response.Status.NOT_FOUND).entity(e.toString()).build();
+                case INVALID_ARGUMENTS:
+                    return Response.status(Response.Status.BAD_REQUEST).entity(e.toString()).build();
+                default:
+                    return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e.toString()).build();
```

Comment:
It can leak internal information to the users. But it can be treated as a vulnerability. Just keep it.

Reply:
In the current design, the internal information is only sent to the frontend server, where the frontend server processes and displays the non-sensitive part to users (e.g. only shows `INTERNALERROR` but not showing the detailed error). We did that to ease the difficulties of debugging both frontend and backend.